### PR TITLE
Deprecate TI cc2650

### PR DESCRIPTION
**Pull request by vanti**
Adding Kconfig settings to inform anyone trying to build for this
platform of its pending deprecation in 2.2.0.

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

File: soc/Kconfig [hunk 1]
```diff
@@ -71,3 +71,11 @@ config SOC_RWDATA_LD
 	  under include/arch/.
 
 endif # ARC || ARM || X86 || NIOS2 || RISCV
+
+config SOC_DEPRECATED
+	string
+	help
+	  This hidden option is set in the soc configuration and indicates
```
Comment (ioannisg) [inline soc/Kconfig]: SoC

Comment (ioannisg) [reply soc/Kconfig]: similar entries below

**Review by ioannisg — COMMENTED** [review]
Shouldn't we have BOARD_DEPRECATED_RELEASE, and SOC_DEPRECATED_RELEASE, to indicated better that the string element refers to the release? Or this terminilogy has been agreed in the API working group?

File: soc/Kconfig (hunk 1 above)
Comment (vanti) [reply soc/Kconfig]: Thanks, fixed.

**Comment by vanti** [comment]
> Shouldn't we have BOARD_DEPRECATED_RELEASE, and SOC_DEPRECATED_RELEASE, to indicated better that the string element refers to the release? Or this terminilogy has been agreed in the API working group?

Not sure about the history on this, but I like your names better. Given I don't see anyone else using this Kconfig option in existing code, I went ahead and renamed them.

**Review by ioannisg — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**